### ISSUE TITLE
pyx problem

### ISSUE THREAD

**Issue by filippoBUO** [cleanlab/cleanlab#111]
Hi,

Once I have the psx, the noise matrix and the inverse noise matrix I compute the pyx with the function compute_pyx. 

The issue I don't get is that for some examples the pyx is so low but CleanLab didn't suggest me this example as possibly having wrong label. Those example have also low psx (<0.5 in binary classification) but since CleanLab didn't detect as Noisy Labels I was expecting  pyx to be greater than 0.5 in binary classification.

Here's the code:

The label for s[1007] is 0.

psx[1007] 
===> array([0.19460601, 0.80539399])

pyx = compute_pyx(psx, noise_matrix, inverse_noise_matrix) 

pyx[1007] 
===> array([0.19995449, 0.80004551])

Since 1007 is not in the returned numpy array of the get_noise_indices function, I was expecting pyx[1007,1]>0.5.

**Comment by cgnorthcutt** [comment]
Hi @filippoBUO 

tl;dr - don't use `compute_pyx` to find label errors (except in very rare circumstances where you have a  near-perfect model). Instead, use `cleanlab.filter.find_label_issues()` for finding label issues in your datasets and `cleanlab.classification.LearningWithNoisyLabels()` for classification with label issues/errors.

Explanation - `compute_pyx` comes from the `latent_algebra` module that we have since moved to `cleanlab.internal.latent_algebra`. It is not intended to be user facing. It's a theoretically correct approach, which, for a perfect model, would uncover the perfect predicted probabilities, but real-world models produce erroneous predicted probabilities and this method is **not** robust to errors in predicted probabilities.
